hsds/servicenode.py

```diff
@@ -37,112 +38,84 @@
     app.router.add_route('PUT', '/datatypes/{id}/attributes/{name}/value', PUT_AttributeValue)
     app.router.add_route('GET', '/datatypes/{id}/acls/{username}', GET_ACL)
-    app.router.add_route('OPTIONS', '/datatypes/{id}/acls/{username}', GET_ACL)
     app.router.add_route('PUT', '/datatypes/{id}/acls/{username}', PUT_ACL)
     app.router.add_route('GET', '/datatypes/{id}/acls', GET_ACLs)
-    app.router.add_route('OPTIONS', '/datatypes/{id}/acls', GET_ACLs)
     app.router.add_route('GET', '/datasets/{id}', GET_Dataset)
-    app.router.add_route('OPTIONS', '/datasets/{id}', GET_Dataset)
     app.router.add_route('GET', '/datasets/', GET_Dataset)
-    app.router.add_route('OPTIONS', '/datasets/', GET_Dataset)
     app.router.add_route('GET', '/datasets', GET_Datasets)
-    app.router.add_route('OPTIONS', '/datasets', GET_Datasets)
     app.router.add_route('DELETE', '/datasets/{id}', DELETE_Dataset)
     app.router.add_route('POST', '/datasets', POST_Dataset)
     app.router.add_route('GET', '/datasets/{id}/shape', GET_DatasetShape)
-    app.router.add_route('OPTIONS', '/datasets/{id}/shape', GET_DatasetShape)
     app.router.add_route('PUT', '/datasets/{id}/shape', PUT_DatasetShape)
     app.router.add_route('GET', '/datasets/{id}/type', GET_DatasetType)
-    app.router.add_route('OPTIONS', '/datasets/{id}/type', GET_DatasetType)
     app.router.add_route('GET', '/datasets/{id}/attributes', GET_Attributes)
-    app.router.add_route('OPTIONS', '/datasets/{id}/attributes', GET_Attributes)
     app.router.add_route('GET', '/datasets/{id}/attributes/{name}', GET_Attribute)
-    app.router.add_route('OPTIONS', '/datasets/{id}/attributes/{name}', GET_Attribute)
     app.router.add_route('DELETE', '/datasets/{id}/attributes/{name}', DELETE_Attribute)
     app.router.add_route('PUT', '/datasets/{id}/attributes/{name}', PUT_Attribute)
     app.router.add_route('GET', '/datasets/{id}/attributes/{name}/value', GET_AttributeValue)
-    app.router.add_route('OPTIONS', '/datasets/{id}/attributes/{name}/value', GET_AttributeValue)
     app.router.add_route('PUT', '/datasets/{id}/attributes/{name}/value', PUT_AttributeValue)
     app.router.add_route('PUT', '/datasets/{id}/value', PUT_Value)
     app.router.add_route('GET', '/datasets/{id}/value', GET_Value)
-    app.router.add_route('OPTIONS', '/datasets/{id}/value', GET_Value)
     app.router.add_route('POST', '/datasets/{id}/value', POST_Value)
     app.router.add_route('GET', '/datasets/{id}/acls/{username}', GET_ACL)
-    app.router.add_route('OPTIONS', '/datasets/{id}/acls/{username}', GET_ACL)
     app.router.add_route('PUT', '/datasets/{id}/acls/{username}', PUT_ACL)
     app.router.add_route('GET', '/datasets/{id}/acls', GET_ACLs)
-    app.router.add_route('OPTIONS', '/datasets/{id}/acls', GET_ACLs)
+
+    # Add CORS to all routes
+    cors = aiohttp_cors.setup(app, defaults={"*": aiohttp_cors.ResourceOptions(allow_credentials=True, expose_headers="*", allow_headers="*",)})
```

Comment:
For now, I hard-corded the config but you might want to handle it here differently.